lib/openstack_query/query_blocks/query_output.py

```diff
@@ -206,7 +206,12 @@ def _parse_forwarded_outputs(
                 # forwarded properties might contain more than one value
                 # then() will keep duplicates so each one in the list will be shunted into an output
                 forwarded_output_dict.update(output_list[0])
-                del output_list[0]
+
+                # a hacky way to prevent one-to-many chaining erroring - keep at least one value
+                # one-to-many/one-to-one will only ever contain one value per grouped_value
+                # many-to-one will contain multiple values per grouped values
+                if len(output_list) > 1:
+                    del output_list[0]
```

Comment:
Can we add a small test to check a list `len ==1` doesn't throw, since this is something small that can be easily missed

Reply:
already exists the following tests test for this: 
`test_parse_forwarded_outputs_one_set` - tests one set of forwarded outputs with a singleton list

and
`test_parse_forwarded_outputs_with_duplicates` - tests one set of forwarded outputs with list > 1

Reply:
Ah but this didn't fail when we deleted the only element? Since nothing failed prior to this bug is my concern

Reply:
I've modified the tests for parse_forwarded_output accordinly to check.
`test_parse_forwarded_outputs_one_to_many` - testing that they aren't deleted (singleton list)
`test_parse_forwarded_outputs_many_to_one` - testing that they are deleted
`test_parse_forwarded_outputs_multiple_sets` - testing when given both one_to_many set and many_to_one set